### PR TITLE
add: feature flag for customize your store task

### PR DESCRIPTION
**Pull request by rjchow**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

- Added feature flag that removes store appearance task and adds customize store task when enabled
- Added task list item header for customize your store task
- Made core profiler feature flagged task removal order invariant, previously it depended on the StoreDetails task being the first in the list

### How to test the changes in this Pull Request:

Note that the feature flag is **disabled even in development.json** because it's not ready for other dev teams to look at

1. Set up WooCommerce, toggle feature flag for 'customize-store' to true
2. Should see the task list as below

image

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/src/Admin/Features/OnboardingTasks/TaskLists.php [hunk 1]
```diff
@@ -41,6 +41,7 @@ class TaskLists {
 	 * @var array
 	 */
 	const DEFAULT_TASKS = array(
+		'CustomizeStore',
```
Comment (ilyasfoo) [inline plugins/woocommerce/src/Admin/Features/OnboardingTasks/TaskLists.php]: In the current state, `CustomizeStore` script will still be imported in the `init_tasks` function. I think we should add check in there as well.

**Review by ilyasfoo — CHANGES_REQUESTED** [review]
Looking good, thanks for the legwork @rjchow! One minor comment about the PHP script still being imported.

File: plugins/woocommerce/src/Admin/Features/OnboardingTasks/TaskLists.php (hunk 1 above)
Comment (rjchow) [reply plugins/woocommerce/src/Admin/Features/OnboardingTasks/TaskLists.php]: as discussed, resolved by adding a comment to clarify the role of DEFAULT_TASKS

Comment (ilyasfoo) [reply plugins/woocommerce/src/Admin/Features/OnboardingTasks/TaskLists.php]: Looks good! Do you intend to remove `CustomizeStore` from the list though?

**Review by ilyasfoo — APPROVED** [review]
Nitpick comment about optionally removing `CustomizeStore` from the defunct list. LGTM anyways!  🚢

**Review by chihsuan — APPROVED** [review]
Great work! LGTM and tested well. 🚀

File: plugins/woocommerce/src/Admin/Features/OnboardingTasks/TaskLists.php (hunk 1 above)
Comment (rjchow) [reply plugins/woocommerce/src/Admin/Features/OnboardingTasks/TaskLists.php]: ok will do 😅

**Pull request merged**